src/components/App/App.js

```diff
@@ -0,0 +1,312 @@
+import React, { useEffect, useState } from "react";
+import { Routes, Route, useNavigate, useLocation } from "react-router-dom";
+
+import { CurrentUserContext } from "../../contexts/CurrentUserContext";
+import { AuthContext } from "../../contexts/AuthContext";
+import ProtectedRoute from "../ProtectedRoute/ProtectedRoute";
+import SigninForm from "../SigninForm/SigninForm";
+import SignupForm from "../SignupForm/SignupForm";
+import PopupTooltip from "../PopupTooltip/PopupTooltip";
+import Header from "../Header/Header";
+import Main from "../Main/Main";
+import Footer from "../Footer/Footer";
+import SavedNews from "../SavedNews/SavedNews";
+import mainApi from "../../utils/MainApi";
+import { getNewsApi } from "../../utils/NewsApi";
+
+function App() {
+  // AUTH states
+  const [currentUser, setCurrentUser] = useState({});
+  const [loggedIn, setLoggedIn] = useState(false);
+  const [conflictErr, setConflictErr] = useState(false);
+
+  // POPUP visibility states:
+  const [isSigninPopupOpen, setSigninPopupOpen] = useState(false);
+  const [isSignupPopupOpen, setSignupPopupOpen] = useState(false);
+  const [isTooltipOpen, setIsTooltipOpen] = useState(false);
+
+  // NEWS states:
+  const [preloader, setPreloader] = useState(false);
+  const [notFound, setNotFound] = useState(false);
+  const [showArticleSection, setShowArticleSection] = useState(false);
+  const [savedArticles, setSavedArticles] = useState([]);
+  const [searchedArticles, setSearchedArticles] = useState([]);
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  //*** AUTH FUNCTIONS ***=================================
+
+  // keep user logged in when token is set:
+  useEffect(() => {
+    (async () => {
+      try {
+        const user = await mainApi.checkToken(localStorage.getItem("jwt").jwt);
```

Comment:
`localStorage.getItem("jwt").jwt`

I don't think you have `("jwt").jwt` in your `localStorage`


I think it should be just `localStorage.getItem("jwt")`